app/src/main/java/com/example/una/CharitySearchListActivity.java

```diff
@@ -42,17 +48,75 @@ protected void onCreate(Bundle savedInstanceState) {
 
         rvCharities = (RecyclerView) findViewById(R.id.rvCharities);
         rvCharities.setAdapter(charitySearchAdapter);
-        rvCharities.setLayoutManager(new LinearLayoutManager(this));
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        rvCharities.setLayoutManager(linearLayoutManager);
         rvCharities.addItemDecoration(new DividerItemDecoration(rvCharities.getContext(), DividerItemDecoration.VERTICAL));
         String query = getIntent().getStringExtra("query");
-        fetchCharities(query);
+
+        scrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
+                // triggered only when new data needs to be appended to the list
+                fetchCharities(query, page);
+            }
+        };
+        // Adds the scroll listener to RecyclerView
+        rvCharities.addOnScrollListener(scrollListener);
+
+        fetchCharities(query, 1);
         setTitle(query);
     }
 
-    private void fetchCharities(String query) {
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // inflate the menu; this adds items to the action bar
+        getMenuInflater().inflate(R.menu.menu_charity_search, menu);
+        final MenuItem searchItem = menu.findItem(R.id.action_search);
+        final SearchView searchView = (SearchView) searchItem.getActionView();
+        // make the search bar cover the title
+        searchView.setMaxWidth(Integer.MAX_VALUE);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                Intent searchListActivity = new Intent(CharitySearchListActivity.this, CharitySearchListActivity.class);
+                searchListActivity.putExtra("query", query);
+                startActivity(searchListActivity);
```

Comment:
wouldn't this create a stack of CharitySearchListActivity if you search multiple terms? Hitting back button will go through all previous search results before going to home screen

Reply:
Yeah, I wasn't sure if we wanted the user to be able to page through all previous search results. I was thinking that it could be useful in certain cases, but I realize it's probably more standard to just repopulate the recycler view instead of creating a stack of CharitySearchListActivity, so I'll work on changing this.